### ISSUE TITLE
Crash at [APMSqliteStore updateStatementForRecord:inTableWithName:columnName:columnValue:error:]_block_invoke SIGABRT (ABORT)

### ISSUE THREAD

**Issue by miniLV** [firebase/firebase-ios-sdk#13024]
### Description

Crashed: com.google.fira.worker: SIGABRT ABORT 0x00000001bdf00bbc 
0 libsystem_kernel.dylib         0x6bbc __pthread_kill + 8 
1 libsystem_pthread.dylib        0xd854 pthread_kill + 208 
2 libsystem_c.dylib              0x1f6ac abort + 124 
3 libsystem_malloc.dylib         0x1b9d8 _malloc_put + 546 
4 libsystem_malloc.dylib         0x1bc40 malloc_zone_error + 96 
5 libsystem_malloc.dylib         0x1007c nanov2_allocate_from_block$VARIANT$mp + 544 
6 libsystem_malloc.dylib         0xf280 nanov2_allocate$VARIANT$mp + 124 
7 libsystem_malloc.dylib         0xf1b4 nanov2_malloc$VARIANT$mp + 60 
8 libsystem_malloc.dylib         0x5efc _malloc_zone_malloc + 148 
9 CoreFoundation                 0x3cd38 __CFStrAllocateMutableContents + 96 
10 CoreFoundation                0x2d650 __CFStringChangeSizeMultiple + 656 
11 CoreFoundation                0x2890c CFStringAppend + 448 
12 CoreFoundation                0x9b58 __CFStringAppendFormatCore + 11020 
13 CoreFoundation                0x59e1c _CFStringAppendFormatAndArgumentsAux2 + 68 
14 CoreFoundation                0x5db64 -[__NSCFString appendFormat:] + 104 
15 MyApp                         0x1592b9c __88-[APMSqliteStore updateStatementForRecord:inTableWithName:columnName:columnValue:error:]_block_invoke + 440 
16 CoreFoundation                0x3d988 __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 16 
17 CoreFoundation                0x981cc -[__NSDictionaryI enumerateKeysAndObjectsWithOptions:usingBlock:] + 176 
18 MyApp                         0x1592890 -[APMSqliteStore updateStatementForRecord:inTableWithName:columnName:columnValue:error:] + 380 
19 MyApp                         0x159062c -[APMSqliteStore updateRecord:inTableWithName:columnName:columnValue:error:] + 272 
20 MyApp                         0x15127a0 -[APMDatabase updateDataType:inTableWithName:columnName:columnValue:error:] + 252 
21 MyApp                         0x15091c4 -[APMDatabase updateDailyCounts:error:] + 560 
22 MyApp                         0x1541010 __57-[APMMeasurement(Event) writeFilteredEventOnWorkerQueue:]_block_invoke + 304 
23 MyApp                         0x15914fc -[APMSqliteStore performTransactionWithError:block:] + 160 
24 MyApp                         0x15084c0 -[APMDatabase performTransaction:] + 40 
25 MyApp                         0x1540eb4 -[APMMeasurement(Event) writeFilteredEventOnWorkerQueue:] + 212 
26 MyApp                         0x1540bc8 -[APMMeasurement(Event) writeEventOnWorkerQueue:] + 1136 
27 MyApp                         0x1540664 -[APMMeasurement(Event) handleEventOnWorkerQueue:] + 488 
28 MyApp                         0x15354a0 -[APMMeasurement logEventOnWorkerQueue:notifyEventListeners:] + 56 
29 MyApp                         0x1535418 -[APMMeasurement logEventOnWorkerQueueWithOrigin:isPublicEvent:name:parameters:timestamp:enabled:ignoreEnabled:ignoreInterceptor:interceptor:addedScreenParameters:] + 524 
30 MyApp                         0x15351ac __151-[APMMeasurement logEventWithOrigin:isPublicEvent:name:parameters:timestamp:enabled:ignoreEnabled:ignoreInterceptor:interceptor:addedScreenParameters:]_block_invoke + 60 
31 MyApp                         0x158860c __51-[APMScheduler scheduleOnWorkerQueueBlockID:block:]_block_invoke + 44 
32 libdispatch.dylib             0x63094 _dispatch_call_block_and_release + 24 
33 libdispatch.dylib             0x64094 _dispatch_client_callout + 16 
34 libdispatch.dylib             0xa73c _dispatch_lane_serial_drain$VARIANT$mp + 644 
35 libdispatch.dylib             0xb1f4 _dispatch_lane_invoke$VARIANT$mp + 408 
36 libdispatch.dylib             0x14ec8 _dispatch_workloop_worker_thread + 632 
37 libsystem_pthread.dylib       0x1e10 _pthread_wqthread + 284 
38 libsystem_pthread.dylib       0x193c start_wqthread + 8

### Firebase SDK Version

10.19

### Xcode Version

15.0

### Installation Method

CocoaPods

### Firebase Product(s)

Analytics, Crashlytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

- Firebase/AnalyticsWithoutAdIdSupport (10.19.0):
    - Firebase/CoreOnly
    - FirebaseAnalytics/WithoutAdIdSupport (~> 10.19.0)
  - Firebase/CoreOnly (10.19.0):
    - FirebaseCore (= 10.19.0)
  - Firebase/Crashlytics (10.19.0):
    - Firebase/CoreOnly
    - FirebaseCrashlytics (~> 10.19.0)
  - FirebaseAnalytics/WithoutAdIdSupport (10.19.0):
    - FirebaseCore (~> 10.0)
    - FirebaseInstallations (~> 10.0)
    - GoogleAppMeasurement/WithoutAdIdSupport (= 10.19.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.11)
    - GoogleUtilities/MethodSwizzler (~> 7.11)
    - GoogleUtilities/Network (~> 7.11)
    - "GoogleUtilities/NSData+zlib (~> 7.11)"
    - nanopb (< 2.30910.0, >= 2.30908.0)
  - FirebaseCore (10.19.0):
    - FirebaseCoreInternal (~> 10.0)
    - GoogleUtilities/Environment (~> 7.12)
    - GoogleUtilities/Logger (~> 7.12)
  - FirebaseCoreExtension (10.24.0):
    - FirebaseCore (~> 10.0)
  - FirebaseCoreInternal (10.24.0):
    - "GoogleUtilities/NSData+zlib (~> 7.8)"
  - FirebaseCrashlytics (10.19.0):
    - FirebaseCore (~> 10.5)
    - FirebaseInstallations (~> 10.0)
    - FirebaseSessions (~> 10.5)
    - GoogleDataTransport (~> 9.2)
    - GoogleUtilities/Environment (~> 7.8)
    - nanopb (< 2.30910.0, >= 2.30908.0)
    - PromisesObjC (~> 2.1)
  - FirebaseInstallations (10.24.0):
    - FirebaseCore (~> 10.0)
    - GoogleUtilities/Environment (~> 7.8)
    - GoogleUtilities/UserDefaults (~> 7.8)
    - PromisesObjC (~> 2.1)
  - FirebaseSessions (10.24.0):
    - FirebaseCore (~> 10.5)
    - FirebaseCoreExtension (~> 10.0)
    - FirebaseInstallations (~> 10.0)
    - GoogleDataTransport (~> 9.2)
    - GoogleUtilities/Environment (~> 7.10)
    - nanopb (< 2.30911.0, >= 2.30908.0)
    - PromisesSwift (~> 2.1)

    

```

**Comment by rizafran** [comment]
Hi @miniLV, your issue seems related to #12962. May I know if your app or any dependency uses sqlite in your code?

**Comment by miniLV** [comment]
> Hi @miniLV, your issue seems related to #12962. May I know if your app or any dependency uses sqlite in your code?

Yes. We utilize `Mixpanel` in our project, which has a dependency on sqlite.

**Comment by paulb777** [comment]
This issue is likely related to linking an sqlite alternative to the system library.
